lib/ansible/playbook/task.py

```diff
@@ -512,6 +591,17 @@ def _get_parent_attribute(self, attr, extend=False, prepend=False):
                         value = self._extend_value(value, parent_value, prepend)
                     else:
                         value = parent_value
+
+                    if (
+                        hasattr(self, '_validate_%s' % attr)
```

Comment:
Moving the action_group resolution and caching back to the Task class is pretty dependent on this bit, and I'm not sure if we want to add it since we've been removing the dependency on these methods where possible rather than adding more support. I can drop this commit (90f3cfe16dc698d57144914f4c82ce29eddd55d8) if it seems like going in the wrong direction, but keeping this contained to the Task class removed the cruft in Base trying to figure out if the play's cache is available.

Reply:
Did end up dropping the commit moving things to task.py ^